### DISCUSSION TITLE
The library of Glyphs is too limited

### DISCUSSION THREAD

**Discussion by elijah286** [ni/labview-icon-editor#165] [General]
I find myself wishing that it was easier to tap into a larger, ideally online and open-source library of glyphs. What ships with the product is just too limited and small so I typically cannot find what I want.

**Comment by crossrulz** [comment]
An option would be to have a library of glyphs in a GitHub repository with a release zip file for all of the glyphs. The Icon Editor could then import directly from the GitHub release files.

**Comment by rtollert** [comment]
I'm not that certain that we are going to get *that* much lift out of open-source icon libraries, because so many of them are designed for a 24-bit, RGBA, high-res world. I sometimes find myself editing some existing glyphs a lot because they were just not really optimized for 32x32 in 8-bit color.

Now: I have been repeatedly bothered by how difficult it is to share custom glyphs between projects. I would really like to be able to maintain a personal icon repository, on a file share or a git repo or something, that all of my LabVIEW installs across multiple machines could sync up to. I think that requirement could also be satisfied by allowing the Icon Editor to just be pointed at a URL of a zip, and extract all of the images in the zip and install them as glyphs. And that might also solve the "use a ZIP of open-source icons" use case.

**Comment by VTRobH** [comment]
The Icons aren't great, true. But what makes this a really bad situation is the naming/organization makes it hard to find something even if you know it's there.

At JET, I've created a subset of the glyphs that aren't functionally any different, however I renamed them all using the following format:
tag tag tag tag
where each tag is a different potential search term for that glyph.
The search function in the icon editor does partial string searching so all of a sudden, type in a relevant term and you get a decent selection of relevant glyphs.

No programming required...

**Comment by gregr-ni** [comment]
I think we could handle the size and color issues with a good re-rendering tool (possibly [Inkscape](https://github.com/inkscape/inkscape)).
The process of converting to a convenient size, like ~1/4 of our icon size, could be automated and either done before publishing or done dynamically on the client.

NI even publishes a couple of sets of icons ([Nimble](https://nimble.ni.dev/storybook/index.html?path=/docs/components-icons--docs), [Fuse](https://ni.github.io/design-system/style/icons/) ) that could be used.

Another potential source is [iconninja](https://www.iconninja.com/silk-icon-sets-3286).
